guacamole/src/main/java/org/apache/guacamole/rest/APIRequest.java

```diff
@@ -42,6 +42,16 @@
     private final Map<String, String[]> parameters;
 
     /**
+     * The remote hostname that initiated the request.
+     */
+    private final String remoteHost;
+
+    /**
+     * The remote ip address that initiated the request.
```

Comment:
Perhaps a bit of a semantic nitpick, but this is not the IP address that initiated the request, because IP addresses can't do things. It's the IP address of the client that initiated the request. Same thing for the hostname above.